src/selectors/events.js

```diff
@@ -50,6 +52,77 @@ export const groupEventsByStartTime = (events: Event[]): EventDays => {
     : sections.days;
 };
 
+const generateRecurringEvent = event => recurrance => {
+  const [recurranceDay, recurrancyMonth, recurranceYear] = recurrance.split(
+    "/"
+  );
+  const [eventStartDate, eventStartTime] = event.fields.startTime[locale].split(
+    "T"
+  );
+  const eventEndTime = event.fields.endTime[locale].split("T")[1];
+  const [eventStartYear, eventStartMonth, eventStartDay] = eventStartDate.split(
+    "-"
+  );
+  const startAndEndAreSameDay = isSameDay(
+    event.fields.startTime[locale],
+    event.fields.endTime[locale]
+  );
+
+  return R.mergeDeepRight(event, {
+    fields: {
+      startTime: {
+        [locale]: formatContentfulDate(
+          recurranceYear,
+          recurrancyMonth,
+          recurranceDay,
+          eventStartTime
+        )
+      },
+      endTime: {
+        [locale]: startAndEndAreSameDay
+          ? formatContentfulDate(
+              recurranceYear,
+              recurrancyMonth,
+              recurranceDay,
+              eventEndTime
+            )
+          : event.fields.endTime[locale]
```

Comment:
Can you explain the behaviour when an `startTime` and `endTime` of an event are not on the same day and it has recurrences? From the code it looks like the recurrence then ends on the `endTime` of the event. What would be an example of an event where this is useful?

Can it happen that the recurrence then starts after it ends? This crashed our app previously for normal events. Maybe we need a guard here.

Reply:
Yea this was a bit of an odd one, and definitely has some edge cases. I figured that if the end time was on same day then we would definitely want to change the day to that of the recurrence.

However I figured if the content editor has selected the event dates over a wide range of days (which they really shouldn't if they are using the recurring days feature) then it would be best not to mess with it.

I guess it is possible that you could end up with an end date that is before the start day of a recurrence with this logic. Maybe I'll tweak the logic so if the end date is the same day _or_ a previous day then modify the day.